tests/monotouch-test/System.Net.Http/MessageHandlers.cs

```diff
@@ -120,6 +120,84 @@ public void TestNSUrlSessionHandlerCookies ()
 			Assert.That (nativeCookies.First (), Is.StringStarting ("cookie=chocolate-chip;"), $"Native Cookie Value");
 			Assert.That (managedCookies.First (), Is.StringStarting ("cookie=chocolate-chip;"), $"Managed Cookie Value");
 		}
+
+		// ensure that we can use a cookie container to set the cookies for a url
+		[Test]
+		public void TestNSUrlSessionHandlerCookieContainer ()
+		{
+			var url = NetworkResources.Httpbin.CookiesUrl;
+			var cookie = new Cookie ("cookie", "chocolte-chip");
+			var cookieContainer = new CookieContainer ();
+			cookieContainer.Add (new Uri (url), cookie);
+
+			string managedCookieResult = null;
+			string nativeCookieResult = null;
+			Exception ex = null;
+			var completed = false;
+
+			TestRuntime.RunAsync (DateTime.Now.AddSeconds (30), async () => {
+				try {
+					var managedHandler = new HttpClientHandler () {
+						AllowAutoRedirect = false,
+						CookieContainer = cookieContainer,
+					};
+					var managedClient = new HttpClient (managedHandler);
+					var managedResponse = await managedClient.GetAsync (url);
+					managedCookieResult = await managedResponse.Content.ReadAsStringAsync ();
+					
+					var nativeHandler = new NSUrlSessionHandler () {
+						AllowAutoRedirect = true,
+						CookieContainer = cookieContainer,
+					};
+					var nativeClient = new HttpClient (nativeHandler);
+					var nativeResponse = await nativeClient.GetAsync (url);
+					nativeCookieResult = await nativeResponse.Content.ReadAsStringAsync ();
+				} catch (Exception e) {
+					ex = e;
+				} finally {
+					completed = true;
+				}
+			}, () => completed);
+
+			Assert.IsNull (ex, "Exception");
+			Assert.IsNotNull (managedCookieResult, "Managed cookies result");
+			Assert.IsNotNull (nativeCookieResult, "Native cookies result");
+			Assert.AreEqual (managedCookieResult, nativeCookieResult, "Cookies");
```

Comment:
My two cents - we should have an assert on the generated `NSUrlRequest` that verifies the cookie header. This could be done by injecting the underlying `NSUrlSession` and capturing the parameter passed to `CreateDataTask `

Not sure if the current structure of the code allows this verification.

Reply:
Or better yet - this test might be a better candidate for an integration test that uses some sort of test server - either in process (https://github.com/WireMock-Net/WireMock.Net) or a live server (e.g. https://httpbin.org/get)

Reply:
The test is ensuring that the same cookies are sent when using the managed handler and the unmanaged. The request from httpbin.org returns a dictionary of cookies + values. So if they are equal, we should be fine.